lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -185,18 +195,18 @@ def __init__(self, *args, **kwargs):
                 self.ver
             )
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(tuple(getattr(self, attr) for attr in _ComputedReqKindsMixin.UNIQUE_ATTRS))
 
-    def __eq__(self, candidate):
+    def __eq__(self, candidate: object) -> bool:
         return hash(self) == hash(candidate)
 
     @classmethod
     def from_dir_path_as_unknown(  # type: ignore[misc]
-            cls,  # type: t.Type[Collection]
-            dir_path,  # type: bytes
-            art_mgr,  # type: ConcreteArtifactsManager
-    ):  # type: (...)  -> Collection
+            cls,
+            dir_path: bytes,
+            art_mgr: ConcreteArtifactsManager,
+    ) -> _ComputedReqKindsMixin:
```

Comment:
Since we do `cls(...)` for building the result I believe we can do `t.Self` to better reflection the returned instance. Has a requirement of Python 3.11 which we easily met.

Reply:
I've confirmed that it helps with typing in other places, thanks! Marking this thread as resolved.